sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala

```diff
@@ -87,4 +87,28 @@ object StringUtils {
     }
     funcNames.toSeq
   }
+
+  class StringRope {
+    private var list = List.empty[String]
```

Comment:
Let's use a `ListBuffer` or an `ArrayBuffer` here. Those have (amortized) constant time appends and do not force you to reverse the collection when building the string.

Reply:
I supposed reverse is relatively cheap O(n). Coping a string would be more expensive than adding element in the head of list inside of the reverse() method.

Reply:
I replaced `List` by `ArrayBuffer`